bittide/src/Bittide/Transceiver/ResetManager.hs

```diff
@@ -171,8 +176,13 @@ resetManager config clk rst tx_init_done rx_init_done rx_data_good =
           (RxWait (countSucc cntr), stats)
 
       -- Monitor link. Move all the way back to 'StartTx' if the link goes down
-      -- for some reason.
-      (Monitor, stats@Statistics{failAfterUps})
+      -- for some reason. On startup `tx_done` can be True while it should not
+      -- be. When the receiver is not connected, this causes the reset manager
+      -- to claim it is done. We therefore also check `tx_done` in `Monitor`,
+      -- going back to `StartTx` when it goes down. We expect this to only
+      -- happen once.
+      (Monitor, stats@Statistics{txRetries, failAfterUps})
+        | not tx_done -> (StartTx, stats{txRetries=satSucc SatBound txRetries})
```

Comment:
Isn't this a `failAfterUp`?

Reply:
Well, it would be if it properly came here through `RxWait`. But when the Rx side is not connected, and both `rx_done` and `rx_good` are `pure True`, did it really fail after being up? Or was it never really up and should it retry Tx?